hudi-spark/src/main/java/org/apache/hudi/keygen/SimpleKeyGenerator.java

```diff
@@ -66,6 +68,14 @@ public HoodieKey getKey(GenericRecord record) {
       partitionPath = partitionPathField + "=" + partitionPath;
     }
 
-    return new HoodieKey(recordKey, partitionPath);
+    return partitionPath;
+  }
+
+  public String getRecordKey(GenericRecord record) {
```

Comment:
did you think if we need to make this an abstract method in KeyGenerator?

Reply:
Yeah this has been discussed already. Please refer to this - https://github.com/apache/incubator-hudi/pull/1433#discussion_r411496396